openstack/sharedfilesystems/v2/shares/results.go

```diff
@@ -101,6 +104,84 @@ type CreateResult struct {
 	commonResult
 }
 
+// SharePage is a pagination.pager that is returned from a call to the List function.
+type SharePage struct {
+	pagination.MarkerPageBase
+}
+
+// NextPageURL generates the URL for the page of results after this one.
+func (r SharePage) NextPageURL() (string, error) {
+	currentURL := r.URL
+	mark, err := r.Owner.LastMarker()
+	if err != nil {
+		return "", err
+	}
+
+	q := currentURL.Query()
+	q.Set("offset", mark)
+	currentURL.RawQuery = q.Encode()
+	return currentURL.String(), nil
+}
+
+// LastMarker returns the last offset in a ListResult.
+func (r SharePage) LastMarker() (string, error) {
```

Comment:
I realize this was copied from `sharenetworks` but I wanted to flag it for a sanity check. 

My interpretation of this is that Gophercloud will request an `offset` of `9223372036854775807` in order to deliberately make an erroneous call which will return no shares. Is this really the only way to determine this? Wouldn't this cause a lot of unnecessary GETs over time?

Reply:
Should be fixed now: `LastMarker` now returns an "invalid marker" (-1) which is recognized by `NextPageURL` which then returns an empty string - this stops the page iteration. Thanks!

Reply:
Thank you for this as well!